host/frontend/host-orchestrator/intercept/js/server_connector.js

```diff
@@ -29,8 +29,16 @@
 // the client loaded.
 // In this case the device id is passed as a parameter in the url.
 export function deviceId() {
-  const urlParams = new URLSearchParams(window.location.search);
-  return urlParams.get('deviceId');
+  // The server connector is loaded from the client.html file, which is loaded
+  // with a path like "/devices/{deviceId}/files/"
+  let path = location.pathname;
+  let beginIdx = path.indexOf("/devices/") + "/devices/".length;
+  let endIdx = path.indexOf("/files/");
+  if (beginIdx < 0 || endIdx < 0) {
```

Comment:
What do you think of splitting `path` on `/` characters and then searching through the resulting array instead?

Reply:
Doing that would break if the device id has `/` characters on it. I don't think we'll allow those in valid device ids, but still it's nice not to have a bug in JS in case we do.
Performance wise I don't think there will be a significant difference: Splitting the entire string by `/` does more work than necessary, but what I did searches for a string from the beginning twice, which is more costly than searching for "devices" first and then for "files". I guess I could change the second search to not start at 0.
Finally, the code would look like this, I'm not sure it's clearer:
```
let pathElements = location.pathname.split('/');
let devIdx = pathElements.indexOf("devices");
let filesIdx = devIdx + 2;
if (devIdx < 0 || filesIdx >= pathElements.length || pathElements[filesIdx] != "files") {
    throw...
}
return pathElements[devIdx + 1];
```
changed the search to not start at 0, let me know if you had other reasons in mind when suggesting this.

Reply:
discussed offline and decided to switch to splitting